spectrum-vrf/src/lottery.rs

```diff
@@ -0,0 +1,47 @@
+use std::{u128, u32};
+
+use elliptic_curve::{CurveArithmetic, FieldBytes};
+use elliptic_curve::point::PointCompression;
+use elliptic_curve::sec1::{FromEncodedPoint, ModulusSize, ToEncodedPoint};
+
+use spectrum_crypto::digest::blake2b256_hash;
+
+use crate::utils::projective_point_to_bytes;
+use crate::vrf::ECVRFProof;
+
+pub fn proof_to_random_number<TCurve: CurveArithmetic + PointCompression>(
+    proof: &ECVRFProof<TCurve>, vrf_range: u32) -> u64
+    where <TCurve as CurveArithmetic>::AffinePoint: FromEncodedPoint<TCurve>,
+          <TCurve as elliptic_curve::Curve>::FieldBytesSize: ModulusSize,
+          <TCurve as CurveArithmetic>::AffinePoint: ToEncodedPoint<TCurve>
+{
+    let c_fb: FieldBytes<TCurve> = proof.c.into();
+    let s_fb: FieldBytes<TCurve> = proof.s.into();
+
+    let proof_bytes = [projective_point_to_bytes::<TCurve>(proof.gamma),
+        c_fb.to_vec(), s_fb.to_vec()].concat();
+    let proof_hash = blake2b256_hash(&proof_bytes);
+    let proof_hash_bytes: [u8; 32] = proof_hash.into();
+    let proof_hash_bytes_vec = proof_hash_bytes.to_vec();
+
+    let proof_num = u64::from_ne_bytes(proof_hash_bytes_vec[0..8].try_into().unwrap());
```

Comment:
Are you sure its safe to project u256 to u64?

Reply:
This is exact the last question that I want to resolve, I didn't found distinct from_bytes methods for u256 in the elliptic_curve lib